landlab/ca/boundaries/hex_lattice_tectonicizer.py

```diff
@@ -473,6 +475,40 @@ def __init__(self, grid=None, node_state=None, propid=None, prop_data=None,
             self.block_layer_dip_angle = block_layer_dip_angle
             self.layer_left_x = layer_left_x
             self.y0_top = y0_top
+            
+    def _setup_links_to_update_after_uplift(self):
+        """Create and store array with IDs of links for which to update
+        transitions after uplift.
+        
+        These are: all active boundary links, plus the lowest non-boundary
+        links, including the next-to-lowest vertical links and those angling
+        that are below them.
+        
+        Examples
+        --------
+        >>> from landlab import HexModelGrid
+        >>> hg = HexModelGrid(6, 6, orientation='vert', shape='rect')
+        >>> lu = LatticeUplifter(grid=hg)
+        >>> lu.links_to_update
+        array([ 8,  9, 11, 12, 13, 14, 15, 16, 18, 19, 20, 21, 22, 24, 25, 26, 30,
+               34, 38, 42, 46, 50, 54, 58, 62, 66, 70, 72, 73, 74, 75, 76, 77, 79,
+               80])
+        >>> hg = HexModelGrid(5, 5, orientation='vert', shape='rect')
+        >>> lu = LatticeUplifter(grid=hg)
+        >>> lu.links_to_update
+        array([ 7, 10, 11, 13, 14, 15, 16, 17, 18, 20, 22, 25, 28, 31, 35, 38, 41,
+               44, 46, 48, 49, 50, 51, 52, 53])
+        """
+        g = self.grid
+        nc = g.number_of_node_columns
+        max_link_id = (3 * (nc - 1) + 2 * ((nc + 1) // 2) + nc // 2
+                       + (nc - 1) // 2)
+        lower_active = logical_and(arange(g.number_of_links) < max_link_id,
+                                   g.status_at_link == 0)
+        boundary = logical_or(g.status_at_node[g.node_at_link_tail] != 0,
+                              g.status_at_node[g.node_at_link_head] != 0)
+        active_bnd = logical_and(boundary, g.status_at_link == 0)
+        self.links_to_update = where(logical_or(lower_active, active_bnd))[0]
```

Comment:
The array returned by `np.where` should be converted to `int` (it will be `int64` on some machines). This can be done with,
```python
self.links_to_update = as_id_array(where(logical_or(lower_active, active_bnd))[0])
```
`as_id_array` is in `landlab/core/utils.py`.